kratos/solving_strategies/strategies/arc_length_strategy.h

```diff
@@ -0,0 +1,693 @@
+    typedef typename BaseType::TBuilderAndSolverType TBuilderAndSolverType;
+    typedef typename BaseType::TDataType TDataType;
+    typedef TSparseSpace SparseSpaceType;
+    typedef typename BaseType::TSchemeType TSchemeType;
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Constructor
+    ArcLengthStrategy(
+        ModelPart& model_part,
+        typename TSchemeType::Pointer pScheme,
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        Parameters& rParameters,
+        int MaxIterations = 30,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = false
+        ) : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme,
+                pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, ReformDofSetAtEachStep, MoveMeshFlag)
+        {
+            mDesiredIterations = rParameters["desired_iterations"].GetInt();
+            mMaxRadiusFactor   = rParameters["max_radius_factor"].GetDouble();
+            mMinRadiusFactor   = rParameters["min_radius_factor"].GetDouble();
+            mInitializeArcLengthWasPerformed = false;
+
+            // we initialize the list of load processes to be taken into account
+            if (rParameters["loads_sub_model_part_list"].size() > 0) {
+                mSubModelPartList.resize(rParameters["loads_sub_model_part_list"].size());
+                mVariableNames.resize(rParameters["loads_variable_list"].size());
+
+                if (mSubModelPartList.size() != mVariableNames.size())
+                    KRATOS_THROW_ERROR( std::logic_error, "For each SubModelPart there must be a corresponding nodal Variable", "")
+
+                for (unsigned int i = 0; i < mVariableNames.size(); i++) {
+                    mSubModelPartList[i] = &( model_part.GetSubModelPart(rParameters["loads_sub_model_part_list"][i].GetString()));
+                    mVariableNames[i] = rParameters["loads_variable_list"][i].GetString();
+                }
+            }
+        }
+
+    /**
+     * @brief Destructor.
+     * @details In trilinos third party library, the linear solver's preconditioner should be freed before the system matrix. We control the deallocation order with Clear().
+     */
+    ~ArcLengthStrategy() override
+    {
+        BaseType::~BaseType();
```

Comment:
You don't need that, default destructors call the base class one, you can put directly = default;, it will be clearer and more valid